### PR TITLE
Update build command to update architecture metadata if necessary

### PR DESCRIPTION
**Pull request by mthalman**
Ensures the build command on ImageBuilder will produce an image whose architecture matches the expected value in case Docker decides to misrepresent it.

### CODE REVIEW

File: Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs [hunk 1]
```diff
@@ -84,6 +92,75 @@ private void BuildImages()
             BuiltTags = BuiltTags.ToArray();
         }
 
+        private void EnsureArchitectureMatches(PlatformInfo platform)
+        {
+            foreach (TagInfo tag in platform.Tags)
```
[the remark this answers is not in the document]
Comment (mthalman) [reply Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs]: Yes, I had the same question when I was implementing it and so I had verified that it needs to be done on a per-tag basis.  Your suggestion sounds good.  I'll work on that.

File: Microsoft.DotNet.ImageBuilder/src/ExecuteHelper.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 using System;
 using System.Diagnostics;
+using System.Text;
```
Comment (MichaelSimons) [inline Microsoft.DotNet.ImageBuilder/src/ExecuteHelper.cs]: Why is this needed when no source changed in this file?

File: Microsoft.DotNet.ImageBuilder/src/DockerHelper.cs [hunk 3]
```diff
@@ -20,20 +20,6 @@ public static class DockerHelper
         public static Architecture Architecture => s_architecture.Value;
         public static OS OS => s_os.Value;
```
Comment (MichaelSimons) [inline Microsoft.DotNet.ImageBuilder/src/DockerHelper.cs]: Shuffling the methods intermixed with code changes makes it rather difficult to review what the concrete changes were made 😸

File: Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs [hunk 4]
```diff
@@ -52,10 +55,11 @@ private void BuildImages()
 
                         // Tag the built images with the shared tags as well as the platform tags.
                         // Some tests and image FROM instructions depend on these tags.
-                        IEnumerable<string> platformTags = platform.Tags
-                            .Select(tag => tag.FullyQualifiedName)
-                            .ToArray();
-                        string tagArgs = GetDockerTagArgs(image, platformTags);
+                        IEnumerable<string> allTags = platform.Tags
```
Comment (MichaelSimons) [inline Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs]: Since you are touching this, now would be a great time to cleanup a historical artifact.  I am pretty certain we don't do anything with the shared tags here anymore.  We used to test off of the shared tags long time ago and even had Dockerfile references to them.  We no longer do this therefore tagging the images with the shared tags is pure overhead.  We don't push them today.  IMO it would be a good time to clean this up.

File: Microsoft.DotNet.ImageBuilder/src/DockerHelper.cs [hunk 5]
```diff
@@ -198,6 +178,47 @@ private static bool ResourceExists(ManagementType type, string filterArg, bool i
...
+            }
+
+            return architecture;
+        }
+
+        private static string ExecuteCommand(
+            string command, string errorMessage, string additionalArgs = null, bool isDryRun = false)
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo(
+                "docker", $"{command} {additionalArgs}");
```
Comment (MichaelSimons) [inline Microsoft.DotNet.ImageBuilder/src/DockerHelper.cs]: NIT - consider making the method call one line.

File: Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs (hunk 4 above)
Comment (MichaelSimons) [reply Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs]: I am retracting this comment.  This suggestion makes sense when running in the context of the official build but not from the build-and-test script perspective.

File: Microsoft.DotNet.ImageBuilder/src/DockerHelper.cs (hunk 3 above)
Comment (mthalman) [reply Microsoft.DotNet.ImageBuilder/src/DockerHelper.cs]: Sorry bout that.  I have an allergic reaction to having private and public methods mixed.

File: Microsoft.DotNet.ImageBuilder/src/ExecuteHelper.cs (hunk 2 above)
Comment (mthalman) [reply Microsoft.DotNet.ImageBuilder/src/ExecuteHelper.cs]: Not needed.  Leftovers from my previous change to the file that was reverted.
